src/features/user/Settings/DeleteProfile.tsx

```diff
@@ -18,25 +19,36 @@ export default function DeleteProfile({}: any) {
   };
   const { handleError } = React.useContext(ErrorHandlingContext);
   const [isUploadingData, setIsUploadingData] = React.useState(false);
-
+  const [isModalOpen, setisModalOpen] = React.useState(false); //true when subscriptions are present
   const [canDeleteAccount, setcanDeleteAccount] = React.useState(false);
 
   const handleDeleteAccount = () => {
     setIsUploadingData(true);
     deleteAuthenticatedRequest('/app/profile', token, handleError).then(
       (res) => {
-        if (res !== 404) {
-          logoutUser(`${process.env.NEXTAUTH_URL}/`);
-        } else {
+        if (res == 400) {
```

Comment:
@prachigarg19 There could be multiple factors that might result into a 400 response. Before opening the Cancel Subscriptions modal, along with 400 response please check the response (i.e error_code, error_type) returned from the server. 
![image](https://user-images.githubusercontent.com/37466979/189659478-62a8801a-35bd-4cd1-99c8-a1b031ad2261.png)